doc/using/overlays.chapter.md

```diff
@@ -2,47 +2,76 @@
+## Applying overlays
 
-## Installing overlays {#sec-overlays-install}
+There are various ways of applying overlays.
 
-The list of overlays can be set either explicitly in a Nix expression, or through `<nixpkgs-overlays>` or user configuration files.
+### Applying overlays when importing Nixpkgs
 
-### Set overlays in NixOS or Nix expressions {#sec-overlays-argument}
+When importing Nixpkgs, a list of overlays can be specified using the `overlays` attribute in the argument:
+```nix
+pkgs = import <nixpkgs> {
+  overlays = [
+    firstOverlay
+    secondOverlay
+  ];
+}
+```
 
-On a NixOS system the value of the `nixpkgs.overlays` option, if present, is passed to the system Nixpkgs directly as an argument. Note that this does not affect the overlays for non-NixOS operations (e.g.  `nix-env`), which are [looked up](#sec-overlays-lookup) independently.
+If the `overlays` argument is not passed, these defaults are used:
+- If the Nix path entry `<nixpkgs-overlays>` exists, we look for overlays at that path, as described below.
 
-The list of overlays can be passed explicitly when importing nixpkgs, for example `import <nixpkgs> { overlays = [ overlay1 overlay2 ]; }`.
+    See the section on `NIX_PATH` in the Nix manual for more details on how to set a value for `<nixpkgs-overlays>.`
+- If one of `~/.config/nixpkgs/overlays.nix` and `~/.config/nixpkgs/overlays/` exists, then we look for overlays at that path, as described below. It is an error if both exist.
 
-NOTE: DO NOT USE THIS in nixpkgs. Further overlays can be added by calling the `pkgs.extend` or `pkgs.appendOverlays`, although it is often preferable to avoid these functions, because they recompute the Nixpkgs fixpoint, which is somewhat expensive to do.
+#### How paths get turned into overlays
+- If the path is a file, then the file is imported as a Nix expression and used as the list of overlays.
 
-### Install overlays via configuration lookup {#sec-overlays-lookup}
+- If the path is a directory, then we take the content of the directory, order it lexicographically, and attempt to interpret each as an overlay by:
 
-The list of overlays is determined as follows.
+  - Importing the file, if it is a `.nix` file.
 
-1.  First, if an [`overlays` argument](#sec-overlays-argument) to the Nixpkgs function itself is given, then that is used and no path lookup will be performed.
+  - Importing a top-level `default.nix` file, if it is a directory.
 
-2.  Otherwise, if the Nix path entry `<nixpkgs-overlays>` exists, we look for overlays at that path, as described below.
+### Applying further overlays on top of an imported Nixpkgs
 
-    See the section on `NIX_PATH` in the Nix manual for more details on how to set a value for `<nixpkgs-overlays>.`
+If you already have an imported Nixpkgs, you can apply further overlays on top using:
+```nix
+pkgs.appendOverlays [
```

Comment:
does this need to be `pkgs = pkgs.appendOverlays` or does it really stand alone?

Reply:
I think that would be a bit confusing then. I just used `pkgs =` in the beginning to signal that it's an imported Nixpkgs.

Maybe we could do something like this to make it clearer though:
```nix
pkgsModified = pkgs.appendOverlays [ ... ]
```